course5/homework6/string.cc

```diff
@@ -0,0 +1,106 @@
+
+#include "string.h"
+
+namespace String {
+
+String::String(const String& copy):
+	mContents(std::make_unique<char[]>(mSize)),
+	mSize(copy.mSize)
```

Comment:
I'm surprised that it works. I would expect mSize to be not initialized yet in line 7.
Isn't valgrind complaining here?